### PR TITLE
Matlab Cheat Sheet

### PR DESCRIPTION
**Pull request by namandistro**
What does your Instant Answer do?

- Show all The Matlab Commands as cheat sheet.

What is the data source for your Instant Answer? (Provide a link if possible)

- https://duck.co/redir/?u=http%3A%2F%2Fwww.dummies.com%2Fhow-to%2Fcontent%2Fmatlab-for-dummies-cheat-sheet.html

Why did you choose this data source?

- One of the Best and even recommnded.

Are there any other alternative (better) data sources?

- Their are changes.

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

- Matlab users, Geeks.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

- Yes, https://duck.co/ideas/idea/6173/matlab-cheat-sheet

Are you having any problems? Do you need our help with anything?

- Nopes.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

- 
screenshot from 2015-08-20 11 41 02

-----
IA Page: https://duck.co/ia/view/matlab_cheat_sheet

### CODE REVIEW

**Comment by moollaza** [comment]
@namandistro thanks for your contribution!

Can you please fill out the [pull request template](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_cheat_sheet.md) and edit your initial comment. Once completed, we'll be sure to review your PR and give you some feedback.

Thanks!

File: share/goodie/cheat_sheets/MATLAB.json [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+            "key":"*",
+            "val":"Multiplies the right operand by the left operand."
+          }, {  
+            "key":"^",
+            "val":"Calculates the exponential value of the right operand by the left operand"
+          }, {  
+            "key":"/",
+            "val":"Divides the left operand by the right operand."
+          }, {  
+            "key":"\",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/MATLAB.json]: This should be `"key":"\\"`

**Comment by mintsoft** [comment]
@namandistro rather than duplicating the entire cheatsheet, can you commit in a symlink for aliases? There should be actual aliases coming pretty soon (there's a PR around) to cheatsheets

**Comment by namandistro** [comment]
@Mintsoft can you elaborate what exactly you mean by that?

**Comment by mintsoft** [comment]
@namandistro At the moment the same cheatsheet is committed in as both `share/goodie/cheat_sheets/MATLAB.json` and `share/goodie/cheat_sheets/matlab.json`

I think now I've looked at it that it's just a mistake and that one of them shouldn't be there (`MATLAB.json`)

**Comment by namandistro** [comment]
@mintsoft That issue was i renamed MATLAB to matlab and i was not knowing that their was to 2 files in there but i made the change

**Comment by namandistro** [comment]
@moollaza i updated the template.

**Comment by mintsoft** [comment]
@namandistro generally looks good; although there's only an implicit mention of the transpose operator `'`; I'd also mention things like `flipud` `fliplr` `diag` `circshift` `rot90` as they're really useful

File: share/goodie/cheat_sheets/matlab.json [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    "id":"matlab_cheat_sheet",
+    "name":"matlab",
+    "description":"MATLAB is an incredibly flexible environment that you can use to perform all sorts of math tasks",
+    "metadata":{
+        "sourceName":"Dummies",
+        "sourceUrl":"http://www.dummies.com/how-to/content/matlab-for-dummies-cheat-sheet.html"
+    },
+    "section_order":[
+        "Common",
+        "MATLAB Common Operator",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/matlab.json]: I think there's an "s" missing? "Common Operator**s**"

File: share/goodie/cheat_sheets/matlab.json [hunk 3]
```diff
@@ -0,0 +1,341 @@
+{
+    "id":"matlab_cheat_sheet",
+    "name":"matlab",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/matlab.json]: Let's capitalize this or switch to uppercase: "Matlab" or "MATLAB". Might also be good to add in  "MathWorks"? e.g. "MathWorks MATLAB"

File: share/goodie/cheat_sheets/matlab.json [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    "description":"MATLAB is an incredibly flexible environment that you can use to perform all sorts of math tasks",
+    "metadata":{
+        "sourceName":"Dummies",
+        "sourceUrl":"http://www.dummies.com/how-to/content/matlab-for-dummies-cheat-sheet.html"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/matlab.json]: Do they support `https`?

File: share/goodie/cheat_sheets/matlab.json [hunk 5]
```diff
@@ -0,0 +1,341 @@
+{
+    "id":"matlab_cheat_sheet",
+    "name":"matlab",
+    "description":"MATLAB is an incredibly flexible environment that you can use to perform all sorts of math tasks",
+    "metadata":{
+        "sourceName":"Dummies",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/matlab.json]: I think we should call it "MATLAB for Dummies". "More at Dummies" just looks/sounds a little odd.

**Comment by moollaza** [comment]
@namandistro @mintsoft we now support and `"aliases": [ ]` array in case you need to list additional triggers.

**Comment by moollaza** [comment]
@namandistro it looks like your screenshot didn't quite finish uploading :) -- I think given the content we should use the `code` or possibly the `terminal` template here.

You can add `"template_type" : "code" OR "terminal"` to change the layout. (there's also "reference" but I don't think its appropriate for this data)

File: share/goodie/cheat_sheets/matlab.json (hunk 4 above)
Comment (namandistro) [inline share/goodie/cheat_sheets/matlab.json]: No, they dont

**Comment by namandistro** [comment]
@mintsoft Can you share me the link.

**Comment by namandistro** [comment]
@moollaza Do have a look now!

File: share/goodie/cheat_sheets/matlab.json [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+        "sourceUrl":"http://www.dummies.com/how-to/content/matlab-for-dummies-cheat-sheet.html"
+    },
+    "section_order":[
+        "Common",
+        "MATLAB Common Operators",
+        "MATLAB Operator Precedence",
+        "Line Plot Styles"
+    ],
+    "sections":{
+        "Common":[
```
Comment (moollaza) [inline share/goodie/cheat_sheets/matlab.json]: Let's make this "Common Commands" it's not very clear what this section is for,

File: share/goodie/cheat_sheets/matlab.json [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+    "id":"matlab_cheat_sheet",
+    "name":"MATLAB",
+    "description":"MATLAB is an incredibly flexible environment that you can use to perform all sorts of math tasks",
+    "metadata":{
+        "sourceName":"Matlab For Dummies",
+        "sourceUrl":"http://www.dummies.com/how-to/content/matlab-for-dummies-cheat-sheet.html"
+    },
+    "section_order":[
+        "Common",
+        "MATLAB Common Operators",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/matlab.json]: Let's remove "MATLAB" from these. We already know that we're dealing with MATLAB

**Comment by moollaza** [comment]
@namandistro I've made a couple more comments on the section names. After that I think this is good :+1:

**Comment by namandistro** [comment]
@moollaza Done, Have a look at it.....

**Comment by moollaza** [comment]
LGTM :+1:

**Comment by moollaza** [comment]
@namandistro congrats! It's live: https://duckduckgo.com/?q=matlab+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by namandistro** [comment]
@Moollaza Their is something weird i can see my name on many of contribution even on this matlab.

**Pull request merged**